torch/_inductor/cpp_builder.py

```diff
@@ -346,25 +346,14 @@ def _get_optimization_cflags() -> List[str]:
         if not config.cpp.enable_floating_point_contract_flag:
             cflags.append("ffp-contract=off")
 
-        if config.is_fbcode():
-            # FIXME: passing `-fopenmp` adds libgomp.so to the generated shared library's dependencies.
-            # This causes `ldopen` to fail in fbcode, because libgomp does not exist in the default paths.
-            # We will fix it later by exposing the lib path.
-            return cflags
-
-        if sys.platform == "darwin":
-            # Per https://mac.r-project.org/openmp/ right way to pass `openmp` flags to MacOS is via `-Xclang`
-            # Also, `-march=native` is unrecognized option on M1
-            cflags.append("Xclang")
-        else:
-            if platform.machine() == "ppc64le":
-                cflags.append("mcpu=native")
-            else:
-                cflags.append("march=native")
-
-        # Internal cannot find libgomp.so
-        if not config.is_fbcode():
-            cflags.append("fopenmp")
+        if sys.platform != "darwin":
+            # https://stackoverflow.com/questions/65966969/why-does-march-native-not-work-on-apple-m1
```

Comment:
@xuhancn Is that true though? https://www.phoronix.com/news/LLVM-Clang-march-native-M1
I think it would be nice to include that flag if possible.

Reply:
@uwu-420 This code is make sense to cover Meta's internal config. Please check the latest main branch.

Reply:
@uwu-420 you can try to add `else` block of `if sys.platform != "darwin":` to improve for Apple devices.

Reply:
@uwu-420 FYI: https://stackoverflow.com/questions/72888632/how-to-check-if-python-is-running-on-an-m1-mac-even-under-rosetta you can submit a PR to optimize for M chips.